internal/host/docker_unix.go

```diff
@@ -55,9 +52,13 @@ func (d Docker) Run(ctx context.Context, cmd host.Cmd) (host.WaitStatus, error)
 	if cmd.Stdin != nil {
 		args = append(args, "--interactive")
 	}
-	args = append(args, []string{"--user", d.User}...)
+
+	dockerConnectionUser := strings.Split(d.Connection, "@")[0]
```

Comment:
this and the other one can be simpler: this is within docker already... and is not a connection anymore.
```suggestion
	user := strings.Split(d.Connection, "@")[0]
```

Reply:
Personal preference here, but it is much easier to have self-explanatory variable names according to it's scope, as this user although a user in the general purpose it is strongly connected to the **docker** scope.

Reply:
however you prefer :-D